lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -219,17 +219,9 @@ pacemakerd_event_cb(pcmk_ipc_api_t *pacemakerd_api,
 
     if (reply->data.ping.status == pcmk_rc_ok) {
         crm_time_t *when = pcmk__copy_timet(reply->data.ping.last_good);
-        char *when_s = crm_time_as_string(when,
-                                          crm_time_log_date
-                                          |crm_time_log_timeofday
-                                          |crm_time_log_with_timezone);
-
-        out->message(out, "pacemakerd-health",
-                     reply->data.ping.sys_from, reply->data.ping.state, NULL,
-                     when_s);
-
+        pcmk__output_pcmkd_state(out, reply->data.ping.state,
+                                 reply->data.ping.sys_from, when);
         crm_time_free(when);
-        free(when_s);
```

Comment:
I think it would make more sense just to modify the "pacemakerd-health" implementations to do this (i.e. pass the time_t instead of when_s to it)

Reply:
Then we lose the way to indicate "unknown time," if we still want that. Currently we use a null string pointer for "unknown time."

We would have to represent an unknown time as either current time or epoch.

Reply:
Hmm, when I wrote that I was thinking "0" would have to mean current time, but I suppose 0 could be used for unknown. We should have already fetched the current time by the time we call the pacemakerd-health message.